From 7f516feffb7c28335de8833d14aa132578cc00a5 Mon Sep 17 00:00:00 2001
From: Artur Wieczorek <artwik@wp.pl>
Date: Sun, 5 Apr 2015 10:16:42 +0200
Subject: Warn if wxUSE_MEDIACTRL is set to 1 and in the same time
 wxUSE_LONLONG is set to 0.

wxMediaCtrl requires wxLongLong datatype and if it is missing then appropriate actions (warning or correction) are executed in chkconf.h.

diff --git a/include/wx/chkconf.h b/include/wx/chkconf.h
index c67c67c..92a58b0 100644
--- a/include/wx/chkconf.h
+++ b/include/wx/chkconf.h
@@ -2247,6 +2247,17 @@
 #   endif
 #endif /* wxUSE_PREFERENCES_EDITOR */
 
+#if wxUSE_MEDIACTRL
+#   if !wxUSE_LONGLONG
+#       ifdef wxABORT_ON_CONFIG_ERROR
+#           error "wxMediaCtrl requires wxUSE_LONLONG"
+#       else
+#           undef wxUSE_LONLONG
+#           define wxUSE_LONLONG 1
+#       endif
+#   endif
+#endif /* wxUSE_MEDIACTRL */
+
 #endif /* wxUSE_GUI */
 
 #endif /* _WX_CHKCONF_H_ */
-- 
1.9.5.msysgit.0

